controllers/numaresourcesoperator_controller.go

```diff
@@ -299,6 +304,34 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 	return daemonSetsNName, nil
 }
 
+func (r *NUMAResourcesOperatorReconciler) deleteUnusedDaemonSets(ctx context.Context, instance *nropv1alpha1.NUMAResourcesOperator, mcps []*machineconfigv1.MachineConfigPool) error {
+
+	var daemonSetList appsv1.DaemonSetList
+	err := r.List(ctx, &daemonSetList, &client.ListOptions{Namespace: instance.Namespace})
+	if err != nil {
+		return err
+	}
+
+	// Generate the names of the DaemonSets that should be running,
+	// one for each MachineConfigPool
+	expectedDaemonSetNames := map[string]struct{}{}
+	for _, mcp := range mcps {
+		generatedDSName := rtestate.GetComponentName(instance.Name, mcp.Name)
+		expectedDaemonSetNames[generatedDSName] = struct{}{}
+	}
+
+	for _, ds := range daemonSetList.Items {
```

Comment:
please filter by owner reference, to be sure that we are only deleting daemon set related to our instance

Reply:
done